Minor improvements

maltezfaria opened the pull request:
Opening this to keep track of the few minor issues we discussed yesterday:

- [x] Renaming `Logger` to `LogInfo` for clarity
- [x] Rename `plot` to `plot_traces` and `dagplot` to `plot_dag`
- [x] ~~Change all `@example` to `jldoctest` in docstrings~~ Forward to the documentation for more information on how to visualise a `LogInfo` object
- [x] Review docstrings to make sure they are up to date new interface
- [x] Use a function instead of a macro for `@log`
- [x] Add a `Base.show` method to logger structure for a better REPL experience (only initial implementation added, see #39 )
- [x] Warn in the documentation about the (mis)use of `@log` macro (i.e. returning before tasks are completed)

maltezfaria committed:
remove outdated `@example` blocks from docstrings

maltezfaria committed:
consistent naming of visualization methods

maltezfaria committed:
consistent naming after change `Logger` --> `LogInfo`

maltezfaria committed:
warn about misuse of `@log`

maltezfaria committed:
rename `Logger` --> `LogInfo`

- Also rename `LoggerInfo` used in plotting to `ExtendedLogInfo`

maltezfaria committed:
use a function for implementing `@log`

maltezfaria committed:
intial `show` method for `LogInfo`

maltezfaria committed:
remove dead code + rename + typos

- general cleanup
- add an `_` to methods which are internal (like `_setscheduler!`)
- fix typos in the docs
- hide output of some `jldoctests`

maltezfaria committed:
update docs and figures

maltezfaria commented:
@ffevotte I think this is good to go on my side. I did not read the documentation again, but I did go over all docstrings and the codebase, making a few minor changes along the way. Do you want to review this, or should I just go ahead and merge it?

ffevotte committed:
A few typos in the documentation

ffevotte commented:
This looks very good to me! I fixed a few more typos in the documentation in the latest commit above.

Also, I added a codecov configuration to allow dropping the overall project coverage by up to 1%, which I think will help to avoid CI failures in cases where we want to remove some dead but documented code.

In any case, provided the tests pass, I think this is good to merge!

maltezfaria committed:
fix typo

maltezfaria commented:
> This looks very good to me! I fixed a few more typos in the documentation in the latest commit above.
> 
> Also, I added a codecov configuration to allow dropping the overall project coverage by up to 1%, which I think will help to avoid CI failures in cases where we want to remove some dead but documented code.
> 
> In any case, provided the tests pass, I think this is good to merge!

Thanks, merging this now!

maltezfaria committed:
update docs after last typo fix
